### DISCUSSION TITLE
Do something on each refetch based off refetchInterval?

### DISCUSSION THREAD

**Discussion by half0wl** [TanStack/query#6746] [Q&A]
I'm refetching data with `refetchInterval` periodically, and I'd like to compare the old data against the newly-fetched data, then do something (flash a notification, etc.) if the data has changed.

Looking at the API docs, there's nothing that allows me to "hook" into the concept of refetches, unless I periodically poll the query store for changes.

What is the best way to accomplish this?

**Comment by TkDodo** [comment]
`useEffect` :)
